libs/api/ebpf_api.cpp

```diff
@@ -1506,85 +1672,21 @@ ebpf_program_load(
 
-        // Create all maps.
-        // TODO: update ebpf_map_definition_t structure so that it contains flag and pinning information.
-        for (int count = 0; count < new_object->maps.size(); count++) {
-            ebpf_map_t* map = _get_next_map_to_create(new_object->maps);
-            if (map == nullptr) {
-                // Any remaining maps cannot be created.
-                result = EBPF_INVALID_OBJECT;
-                goto Done;
-            }
-
-            ebpf_handle_t inner_map_handle = (map->inner_map) ? map->inner_map->map_handle : ebpf_handle_invalid;
-            result = _create_map(map->name, &map->map_definition, inner_map_handle, &map->map_handle);
-            if (result != EBPF_SUCCESS) {
-                goto Done;
-            }
-            map->map_fd = _create_file_descriptor_for_handle(map->map_handle);
-        }
-
-        for (auto& program : new_object->programs) {
-            result = _create_program(
-                program->program_type, file_name, program->section_name, program->program_name, &program->handle);
-            if (result != EBPF_SUCCESS) {
-                goto Done;
-            }
-
-            program->fd = _create_file_descriptor_for_handle(program->handle);
-
-            // populate load_info.
-            load_info.file_name = const_cast<char*>(file_name);
-            load_info.section_name = const_cast<char*>(program->section_name);
-            load_info.program_name = const_cast<char*>(program->program_name);
-            load_info.program_type = program->program_type;
-            load_info.program_handle = reinterpret_cast<file_handle_t>(program->handle);
-            load_info.execution_type = execution_type;
-            load_info.byte_code = program->byte_code;
-            load_info.byte_code_size = program->byte_code_size;
-            load_info.execution_context = execution_context_kernel_mode;
-            load_info.map_count = (uint32_t)new_object->maps.size();
-
-            if (load_info.map_count > 0) {
-                for (auto& map : new_object->maps) {
-                    fd_t inner_map_original_fd = (map->inner_map) ? map->inner_map->original_fd : ebpf_fd_invalid;
-                    handle_map.emplace_back(
-                        map->original_fd, inner_map_original_fd, reinterpret_cast<file_handle_t>(map->map_handle));
-                }
-
-                load_info.handle_map = handle_map.data();
-            }
-
-            result = ebpf_rpc_load_program(&load_info, log_buffer, &error_message_size);
-            if (result != EBPF_SUCCESS) {
-                goto Done;
-            }
-        }
-
-        for (auto& program : new_object->programs) {
-            _ebpf_programs.insert(std::pair<ebpf_handle_t, ebpf_program_t*>(program->handle, program));
-        }
-        for (auto& map : new_object->maps) {
-            _ebpf_maps.insert(std::pair<ebpf_handle_t, ebpf_map_t*>(map->map_handle, map));
-        }
+        new_object->loaded = true;
```

Comment:
`ebpf_object_load()` is also setting this to true.

Reply:
Right, removed duplicate